controllers/operands/kubevirtConsolePlugin.go

```diff
@@ -0,0 +1,218 @@
+
+	spec := corev1.ServiceSpec{
+		Ports:    servicePorts,
+		Selector: labelSelect,
+	}
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kvUIPluginName,
+			Labels:    getLabels(hc, hcoutil.AppComponentDeployment),
+			Namespace: hc.Namespace,
+		},
+		Spec: spec,
+	}
+}
+
+func NewKvConsolePlugin(hc *hcov1beta1.HyperConverged) *consolev1alpha1.ConsolePlugin {
+	return &consolev1alpha1.ConsolePlugin{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   kvUIPluginName,
+			Labels: getLabels(hc, hcoutil.AppComponentDeployment),
+		},
+		Spec: consolev1alpha1.ConsolePluginSpec{
+			DisplayName: "Kubevirt Console Plugin",
+			Service: consolev1alpha1.ConsolePluginService{
+				Name:      kvUIPluginName + "-service",
+				Namespace: hc.Namespace,
+				Port:      int32(hcoutil.UiPluginServerPort),
+				BasePath:  "/",
+			},
+		},
+	}
+}
+
+func newConsolePluginHandler(Client client.Client, Scheme *runtime.Scheme, required *consolev1alpha1.ConsolePlugin) Operand {
+	h := &genericOperand{
+		Client:              Client,
+		Scheme:              Scheme,
+		crType:              "ConsolePlugin",
+		removeExistingOwner: false,
+		hooks:               &consolePluginHooks{required: required},
+	}
+
+	return h
+}
+
+type consolePluginHooks struct {
+	required *consolev1alpha1.ConsolePlugin
+}
+
+func (h consolePluginHooks) getFullCr(_ *hcov1beta1.HyperConverged) (client.Object, error) {
+	return h.required.DeepCopy(), nil
+}
+
+func (h consolePluginHooks) getEmptyCr() client.Object {
+	return &consolev1alpha1.ConsolePlugin{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: h.required.Name,
+		},
+	}
+}
+
+func (h consolePluginHooks) getObjectMeta(cr runtime.Object) *metav1.ObjectMeta {
+	return &cr.(*consolev1alpha1.ConsolePlugin).ObjectMeta
+}
+
+func (h consolePluginHooks) reset() { /* no implementation */ }
+
+func (h consolePluginHooks) updateCr(req *common.HcoRequest, Client client.Client, exists runtime.Object, _ runtime.Object) (bool, bool, error) {
+	found, ok := exists.(*consolev1alpha1.ConsolePlugin)
+	updated := false
+
+	if !ok {
+		return false, false, errors.New("can't convert to ConsolePlugin")
+	}
+
+	if !reflect.DeepEqual(found.Spec, h.required.Spec) {
+		if req.HCOTriggered {
+			req.Logger.Info("Updating existing ConsolePlugin to new opinionated values", "name", h.required.Name)
+		} else {
+			req.Logger.Info("Reconciling an externally updated ConsolePlugin to its opinionated values", "name", h.required.Name)
+		}
+		hcoutil.DeepCopyLabels(&h.required.ObjectMeta, &found.ObjectMeta)
+		h.required.Spec.DeepCopyInto(&found.Spec)
+		err := Client.Update(req.Ctx, found)
+		if err != nil {
+			return false, false, err
+		}
+		updated = true
+	}
+
+	// Enable console plugin for kubevirt if not already enabled
+	consoleKey := client.ObjectKey{Namespace: hcoutil.UndefinedNamespace, Name: "cluster"}
+	consoleObj := &operatorv1.Console{}
+	err := Client.Get(req.Ctx, consoleKey, consoleObj)
```

Comment:
When it is being run in the context of the unit-test, the `consoles.operator.openshift.io` resource could not be fetched, and I'm getting:
```
"consoles.operator.openshift.io \"cluster\" not found"
```
Although I'm adding the resource in:
https://github.com/kubevirt/hyperconverged-cluster-operator/pull/1809/files#diff-7e657fcfc15ca5a4e8518504c3b763bf67b211a6a3577c59881b37e557377c96R195-R197

```
	expectedConsoleConfig := &operatorv1.Console{
		TypeMeta: metav1.TypeMeta{
			Kind:       "Console",
			APIVersion: "operator.openshift.io/v1",
		},
		ObjectMeta: metav1.ObjectMeta{
			Name: "cluster",
		},
	}
	res.consoleConfig = expectedConsoleConfig
```

Suggestions?